File: LaboratoryWork/Laboratory_2/C#/task3/task3.cs
using System;

class Task3 {
    static bool is_number(int a) {
        /* Проверяет, что это число, а не цифра. */
        return (a / 10 != 0);
    }

    static int cnt_odd_digits(int a) {
        /* Считает количество нечетных цифр в числе. */
        int res = 0;
        a = Math.Abs(a);

        while (a > 0) {
            res += (a % 2);
            a /= 10;
        }
        return res;
    }

    static void Main(string[] args) {
        int n;
        Console.Write("Введите количество чисел: ");

        n = int.Parse(Console.ReadLine());

        for (int i = 0; i < n; i++) {
            int num;
            Console.Write("Введите " + (i + 1) + " число: ");
            num = int.Parse(Console.ReadLine());

            int cnt_odd = 0;
            if (is_number(num)) {
                cnt_odd = cnt_odd_digits(num);
            } else {
                Console.WriteLine(num + " - это цифра!");
            }

            Console.WriteLine("Количество нечетных цифр: " + cnt_odd);
        }
    }
}

File: LaboratoryWork/Laboratory_7/Laboratory_2/TypeScript/task3.ts
import * as readlineSync from 'readline-sync';

function is_number(a: number): boolean {
    /* Проверяет, что это число, а не цифра. */
    return (Math.floor(a / 10) != 0);
}


function cnt_odd_digits(a: number): number {
    /* Считает количество нечетных цифр в числе. */
    let res: number = 0;
    a = Math.abs(a);

    while (a > 0) {
        res += (a % 2);
        a = Math.floor(a / 10);
    }

    return res;
}


function main() {
    let n: number = parseInt(readlineSync.question("Введите количество чисел: "));

    for (let i: number = 0; i < n; i++) {
        let num: number = parseInt(readlineSync.question("Введите " + (i + 1) + " число: "));
        let cnt_odd: number = 0;

        if (is_number(num)) {
            cnt_odd = cnt_odd_digits(num);
        } else {
            console.log(num + " - это цифра!");
        }

        console.log("Количество нечетных цифр: " + cnt_odd);
    }
}


main();